src/model.spec.ts

```diff
@@ -320,6 +330,16 @@ describe('isValidPurposes', () => {
         invalidKeysPurposesListArbitrary,
         (invalidPurposes) => !isValidPurposes(invalidPurposes)));
   });
+
+  test('does not accept an object missing required purposes', () => {
+    const removeProperty =
+        (propKey: any, {[propKey]: propValue, ...rest}: any) => rest;
```

Comment:
This is nice

Reply:
Only works with the no strict annotation though, alas. I couldn't work out how to do it with proper types :-(